R/translate-sql-odbc.R

```diff
@@ -35,7 +50,79 @@ base_odbc_agg <- sql_translator(
 base_odbc_win <- sql_translator(
   .parent = base_win,
   n             = function() sql("COUNT(*)"),
-  count         = function() sql("COUNT(*)")
+  count         = function() sql("COUNT(*)"),
+
+  # Window functions in this variant use field that its being
+  # 'windowed' as the secondary ORDER BY instead of the field
+  # from win_current_order().  In the tests, having an ORDER BY
```

Comment:
I ran the query from your previous comment against two databases.  The query failed when I run it against MSSQL, but it succeeds when running it against Impala.

MS SQL error:

```
Error in new_result(connection@ptr, statement) : 
  nanodbc/nanodbc.cpp:1344: HY000: The ORDER BY clause is invalid in views, 
inline functions, derived tables, subqueries, and common table expressions, 
unless TOP, OFFSET or FOR XML is also specified., Statement(s) could not be 
prepared.,  
Failed to execute SQL chunk
```
It looks like moving it to the MS SQL translation is the way to go, is that ok?